staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -165,6 +165,25 @@ __kubectl_get_resource()
     __kubectl_parse_get "${nouns[${#nouns[@]} -1]}"
 }
 
+__kubectl_get_resources()
+{
+    if [[ ${#nouns[@]} -eq 0 ]]; then
+      local kubectl_out HEAD TAIL
+      if kubectl_out=$(__kubectl_debug_out "kubectl api-resources $(__kubectl_override_flags) -o name --cached --request-timeout=5s --verbs=get"); then
```

Comment:
If possible, do you mind to avoid duplicate code with kubectl_get_resource?

Reply:
done